agent-control/src/version_checker/handler.rs

```diff
@@ -0,0 +1,56 @@
+use crate::agent_control::agent_id::AgentID;
+use crate::sub_agent::error::SubAgentError;
+use crate::version_checker::AgentVersion;
+use opamp_client::StartedClient;
+use opamp_client::opamp::proto::{AnyValue, KeyValue, any_value};
+
+/// This method request the AgentDescription from the current opamp client and, updates or add the
+/// field from agent version to be sent to opamp server
+pub fn on_version<C>(
```

Comment:
We already had the `on_version` name but I wonder if it would make sense to update it in order to describe what it does (instead of matching the event that triggers it). Something like `update_version_on_agent_description` maybe?

Reply:
Renamed in https://github.com/newrelic/newrelic-agent-control/pull/1563/commits/6211e6d8f77a55c5d0fe368a2edf001078f995bd, LMKWYT!